src/main/java/org/jabref/logic/util/io/FileUtil.java

```diff
@@ -524,4 +524,66 @@ public static boolean detectBadFileName(String fileName) {
     public static boolean isCharLegal(char c) {
         return Arrays.binarySearch(ILLEGAL_CHARS, c) < 0;
     }
+
+    public static boolean renameLinkedFile(LinkedFile linkedFile, BibEntry entry, String fileNamePattern, List<Path> fileDirectories, FilePreferences filePreferences, BibDatabaseContext bibDatabaseContext) {
```

Comment:
Is this method duplicated from somewhere else in the codebase? I'm sure we have already implemented this.

Reply:
We have another implementation at `LinkedFileHandler` that differs slightly from this one. Can we refactor this to remove the duplication? Perhaps we can combine both implementations and change dependents of LinkedFileHandler and LinkedFileAutoRenamer to depend on the same implementation.

Reply:
Did this as well!